src/Diagnostics.DataProviders/DataProviders/KustoDataProvider.cs

```diff
@@ -288,5 +288,18 @@ public async Task<string> GetAggHiPerfClusterNameByStampAsync(string stampName)
                 return null;
             }
         }
+
+        public async Task<DataTable> ExecuteQueryOnHiPerfClusterWithBackup(string aggQuery, string backupQuery, string stampName, string requestId = null, string operationName = null) 
+        {
+            string cluster = await GetAggHiPerfClusterNameByStampAsync(stampName);
+            if (cluster == null)

```

Comment:
nit: Can we do !string.IsNullOrWhiteSpace(cluster)?

```suggestion
            if (!string.IsNullOrWhiteSpace(cluster))

```

Reply:
sure, fixe4d